### PR TITLE
New Admin Vendor Manager Page

### LINKED ISSUE #629: Vendors Page
**Overview**
Create a dedicated page for vendor management instead of via the user's page. This page is a new vendor management page which will list all vendors and allow you to do the following. 

- Add / Remove vendors 
- Approve / Deny vendors 
- Enable or disable a vendor 
- Edit vendor and all related settings/capabilities 

**Describe the solution you'd like**
Clean vendor page and vendor edit page to easily manage vendors from the WordPress Admin. 

**Tasks** 

- [X] Add new Vendors Page Under WC Vendors > Vendors 
- [ ] Vendor list table will have the following columns (all/vendors/pending)
	- [ ] Username (actions column [Edit|delete])
	- [ ] Shop Name
	- [ ] email address 
	- [ ] role
	- [ ] enabled (slide toggle)
	- [ ] bulk actions (deny, approve, delete, disable, enable)
	- [ ] search box (username, shop name, email) 
- [ ] Edit vendor screen provides a single interface to edit everything related to the vendor, settings, capabilities, commission overrides

### PR DESCRIPTION
**Pull request by Mrlunsuper**
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #629 .

### How to test the changes in this Pull Request:

1. Login as Admin account
2. See submenu Vendors on WC Vendors menu.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Review by digitalchild — CHANGES_REQUESTED** [review]
Please see comments below

**Review by mahlamusa — CHANGES_REQUESTED** [review]
Please see comments.

- There are multiple uses of incorrect and inconsistent text domain: replace wcv-vendor, wc-vendor with the correct text domain `wc-vendors`
- Avoid concatinating variables into strings, use 'sprintf' function and placeholders instead
- The delete row action does not work
- The toggle button does not work. When you enable/approve a vendor and reload the page, the switch must be on for enabled vendors

File: src/Admin/Assets.php [hunk 1]
```diff
@@ -29,7 +29,8 @@ public function init_hooks() {
 	 * Enqueue the styles
 	 */
 	public function admin_styles() {
-
+		wp_register_style('wcv_vendors_style',WCV_PLUGIN_URL.'assets/src/scss/vendors-css.css');
```
Comment (mahlamusa) [inline src/Admin/Assets.php]: Enqueue styles from `WCV_PLUGIN_URL . 'assets/css/vendors-css.css`, the src folder is for development only.

Use the build tools to build the css int the `assets/css` and enqueue from there.

File: src/Admin/Assets.php [hunk 2]
```diff
@@ -48,6 +49,12 @@ public function admin_scripts() {
 			wp_register_script( 'wcv_quick-edit', WCV_PLUGIN_URL . 'assets/js/admin/product-quick-edit.js', array( 'jquery' ), WCV_VERSION, true );
 			wp_enqueue_script( 'wcv_quick-edit' );
 		}
+
+		// Vendor Management screen.
+		if ( in_array( $screen_id, array( 'wc-vendors_page_wcv-vendors-management' ), true ) ) {
+			wp_register_script( 'wcv_vendor-edit', WCV_PLUGIN_URL . 'assets/src/js/admin/vendor-edit.js', array( 'jquery' ), WCV_VERSION, true );
```
Comment (mahlamusa) [inline src/Admin/Assets.php]: use scripts from assets/js folder, src folder is for development only

File: src/Admin/Menus.php [hunk 3]
```diff
@@ -168,6 +173,41 @@ public function commission_screen_options() {
 
 		return true;
 	}
+	public function vendors_screen_option() {
+
+		$option = 'per_page';
+		$args   = [
+			'label'   => __('Vendors','wc-vendor'),
```
Comment (mahlamusa) [inline src/Admin/Menus.php]: I still see instances of 'wc-vendor' as text domain, it should be 'wc-vendors'

**Review by mahlamusa — CHANGES_REQUESTED** [review]

**Comment by mahlamusa** [comment]
The toggle to enable a vendor is still not working, I think this is because the ajaxurl is localized as ajax_url and it must be used like so:
```$.ajax({
	method: 'POST',
	url: wcv_ajax_obj.ajax_url,
	...```

Also the row actions in the table are still not firing.

File: src/Admin/Menus.php (hunk 3 above)
Comment (Mrlunsuper) [reply src/Admin/Menus.php]: Change text domain to wc-vendors

File: src/Admin/Assets.php (hunk 2 above)
Comment (Mrlunsuper) [reply src/Admin/Assets.php]: add css and js folder and change page slug

File: src/Admin/Assets.php (hunk 1 above)
Comment (Mrlunsuper) [reply src/Admin/Assets.php]: added css folder and build css file from scss file

**Review by digitalchild — CHANGES_REQUESTED** [review]
The code formatting does not look to be to WordPress standards, please ensure that you're using the PHPCS WordPress sniffs when submitting code.

**Review by mahlamusa — APPROVED** [review]

**Review by digitalchild — CHANGES_REQUESTED** [review]
Looks good. The only thing is the Approve / Deny row actions should only show for pending vendors.

**Review by digitalchild — APPROVED** [review]

**Pull request merged**